<launch>
    <!-- Defining arguments for the mur launch --> 
    <arg name="controllers_started" default="joint_state_controller"/>
    <arg name="controllers_stopped" default="empty"/>  <!--Must not be empty-->
    <arg name="pose_filename" default="$(find mur_launch_hardware)/poses/my_poses.yaml"/>
    <arg name="tf_prefix" default="mur"/>
 
   
    <!-- Bringup the base of the mur -->
    <include file="$(find mur_launch_hardware)/launch/mur_base.launch" pass_all_args="true" ns="ur">      


    <!-- Bringup the ur -->   
    <!-- Fixing missing tf_prefix forwarding to the ur packages -->
    <!-- <param name="mur/ur/robot_state_publisher/tf_prefix" value="$(arg tf_prefix)/ur"/>    -->
    
    <node pkg="tf2_ros" type="static_transform_publisher" name="mur_transform" args="0.1734 -0.1026 0.4 3.141592653589793 0 0 mur/mir/base_link base_link" output="screen"/>
    <param name="mur/ur/robot_state_publisher/tf_prefix" value=""/>

    <!-- <include file="$(find ur_launch_hardware)/launch/ur.launch" ns="mur/ur" pass_all_args="true">
        <arg name="tf_prefix" value="$(arg tf_prefix)/ur/"/>
    </include>  -->

    <include file="$(find ur_launch_hardware)/launch/ur.launch"  pass_all_args="true">
        <arg name="tf_prefix" value=""/>
    </include>

    <include file="$(find ur16e_moveit_config)/launch/move_group.launch"/>
    
    <node pkg="mur_sb" type="current_pos_pub.py" name="current_pos_pub" output="screen"/>


    <arg name="node_start_delay" default="20.0" />
    <node pkg="mur_controller" type="load_controller.py" name="load_controller" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
        <param name="start_controllers" value="joint_group_vel_controller"/>
        <param name="stop_controllers" value="twist_controller"/>
    </node>

</launch>